lib/terraforming/resource/ec2.rb

```diff
@@ -96,6 +96,10 @@ def in_vpc?(instance)
           (instance.subnet_id && instance.subnet_id != "" && instance.security_groups.empty?)
       end
 
+      def associate_public_ip?(instance)
+        (instance.public_ip_address.to_s.empty?) ? false : true
```

Comment:
hmm... `String#empty?` returns boolean value, `true` or `false`, so this can be written more simply,

```ruby
!instance.public_ip_address.to_s.empty?
```